plugins/mapper-size/src/main/java/org/elasticsearch/index/mapper/size/SizeFieldMapper.java

```diff
@@ -50,7 +51,7 @@ public SizeFieldMapper build() {
 
     private static class SizeFieldType extends NumberFieldType {
         SizeFieldType() {
-            super(NAME, NumberType.INTEGER);
+            super(NAME, NumberType.INTEGER, true, true, true, false, null, Collections.emptyMap(), null, false, null, null);
```

Comment:
@romseygeek the specific change is here: https://github.com/elastic/elasticsearch/blob/547c8327b22b395293d02354e6c85d1f46995374/server/src/main/java/org/elasticsearch/search/fetch/subphase/StoredFieldsPhase.java#L80

Previously, stored fields addition didn't actually check the field type, but you added it. 

It seems correct to me to only add stored fields that are stored...but, I am not sure. It is technically a behavior change.